src/datagrid/DataGrid.tsx

```diff
@@ -167,50 +183,60 @@ export enum GridRowType {
  * @param {allColumns} column data
  * @param {allRows} row data
  * @param {itemText} label to display for all items
+ * @param {pagination} pagination data
  */
 type DataGridState = {
     selectAll: boolean;
     allColumns: DataGridColumn[];
     allRows: DataGridRow[];
     itemText: string;
+    pagination?: DataGridPaginationState;
+};
+
+type DataGridPaginationState = {
+    currentPage: number;
+    pageSize: number;
+    totalItems: number;
+    firstItem: number;
+    lastItem: number;
+    totalPages: number;
+    pageSizes: number[];
 };
 
 /**
  * DataGrid Componnet :
  * Displays data in grid format
  */
 export class DataGrid extends React.PureComponent<DataGridProps, DataGridState> {
-    // Default props for datagrid
-    static defaultProps = {
-        pagination: false,
-    };
+    private pageIndexRef = React.createRef<HTMLInputElement>();
 
     // Initial state of datagrid
     state: DataGridState = {
         selectAll: false,
         allColumns: this.props.columns,
         allRows: this.props.rows !== undefined ? this.props.rows : [],
         itemText: this.props.itemText !== undefined ? this.props.itemText : "items",
+        pagination:
+            this.props.pagination !== undefined
+                ? {
+                      currentPage:
+                          this.props.pagination.currentPage !== undefined ? this.props.pagination.currentPage : 1,
+                      pageSize: this.props.pagination.pageSize !== undefined ? this.props.pagination.pageSize : 10,
+                      totalItems: this.props.pagination.totalItems !== undefined ? this.props.pagination.totalItems : 0,
+                      pageSizes:
+                          this.props.pagination.pageSizes !== undefined
+                              ? this.props.pagination.pageSizes
+                              : [10, 20, 50, 100],
+                      firstItem: 0,
+                      lastItem: 0,
+                      totalPages: 0,
```

Comment:
totalPages default value should be one, since we are displaying the datagrid even if it is empty

Reply:
done